[CI] WriteLoadConstraintMonitorTests testRerouteIsNotCalledWhenNoNodesAreHotSpotting failing

**Build Scans:**
- [elasticsearch-periodic #11532 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/7hsheb5ebwtts)
- [elasticsearch-pull-request #107137 / part-1](https://gradle-enterprise.elastic.co/s/4zvxb56vn3rpw)
- [elasticsearch-pull-request #106494 / part-1](https://gradle-enterprise.elastic.co/s/yc6rbj4ztr4f2)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitorTests.testRerouteIsNotCalledWhenNoNodesAreHotSpotting" -Dtests.seed=5EDE1F35119342F2 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=en-ES -Dtests.timezone=Asia/Thimphu -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRerouteIsNotCalledWhenNoNodesAreHotSpotting'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected to see don't reroute due to no nodes hot-spotting but did not
Expected: <0L>
     but: was <1L>
```

**Issue Reasons:**
- [main] 3 failures in test testRerouteIsNotCalledWhenNoNodesAreHotSpotting (0.4% fail rate in 749 executions)
- [main] 2 failures in step part-1 (0.5% fail rate in 409 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.5% fail rate in 406 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -303,9 +303,6 @@ tests:
 - class: org.elasticsearch.xpack.ilm.actions.DownsampleActionIT
   method: testILMWaitsForTimeSeriesEndTimeToLapse
   issue: https://github.com/elastic/elasticsearch/issues/138843
-- class: org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitorTests
-  method: testRerouteIsNotCalledWhenNoNodesAreHotSpotting
-  issue: https://github.com/elastic/elasticsearch/issues/138924
 - class: org.elasticsearch.xpack.ml.integration.DatafeedJobsIT
   method: testStopLookback_closeJobFalse
   issue: https://github.com/elastic/elasticsearch/issues/139024

